File: FirstNotRepeatingChar.cpp
/*
题目描述:在一个字符串(1<=字符串长度<=10000，全部由字母组成)中找到第一个只出现一次的字符,并返回它的位置
*/

#include <string>
#include <vector>

using namespace std;

class Solution {
public:
	int FirstNotRepeatingChar(string str) {
		if (str.empty() || str.size() == 0)
			return -1;
		vector<int> vec(256, 0);
		for (char c : str)
			vec[int(c)]++;
		for (int i = 0; i < str.size(); i++)
			if (vec[int(str[i])] == 1)
				return i;
		return -1;
	}
};

File: FirstNotRepeatingChar.py
# -*- coding:utf-8 -*-

#题目描述:在一个字符串(1<=字符串长度<=10000，全部由字母组成)中找到第一个只出现一次的字符,并返回它的位置

class Solution:
	def FirstNotRepeatingChar(self, s):
		if not s:
			return -1
		for i,ch in enumerate(s):
			if s.count(ch)==1:
				return i

	def FirstNotRepeatingChar(self,s):
		index=0
		if len(s)==0:
			return -1
		else:
			for x in s:
				if s.count(x)==1:
					return index
				index+=1